src/example-types/metricGroups/pivot.js

```diff
@@ -254,6 +254,30 @@ let createPivotScope = (node, schema, getStats) => {
     }
   }
 
+  let filterGroupRanges = ({ drilldownKey, group }) => {
+    if (group.type === 'dateRanges' && drilldownKey) {
+      let [from, to] = _.map(
+        x => new Date(x).toISOString(),
+        _.split(/(?<=Z)-/, drilldownKey)
+      )
+      group.ranges = _.filter(
+        x =>
+          new Date(x.from).toISOString() === from &&
+          new Date(x.to).toISOString() === to,
+        group.ranges
+      )
+    }
+    if (group.type === 'numberRanges' && drilldownKey) {
+      let [from, to] = _.map(parseFloat, _.split('-', drilldownKey))
+      group.ranges = _.filter(
+        x => parseFloat(x.from) === from && parseFloat(x.to) === to,
+        group.ranges
+      )
+    }
+
+    return group
+  }
```

Comment:
We should not have type-specific code inside of `pivot` - these should either just live in `drilldown` or, if necessary, a new hook per type

Reply:
Ok I'll try to find a more appropriate way to solve this issue 👍

Reply:
Maybe we should just define specific `filterGroupRanges`  in `dateRangesGroupStats.js`  and `numberRangesGroupStats.js`. And then do
```
let filterGroupRanges = lookupTypeProp(_.identity, 'filterGroupRanges', group.type)
group = filterGroupRanges(group, drilldownKey)
let build = lookupTypeProp(_.identity, 'buildGroupQuery', group.type)
return build(group, await children, groupingType, schema, getStats)
```